src/as3hx/Writer.hx

```diff
@@ -1276,7 +1276,14 @@ class Writer
                         f(def.el[def.el.length - 1], def.el);
                     }
                 }
-                newCases = loopCases(cases.slice(0), def == null ? null : def.el.slice(0), testVar, newCases);
+
+                if (def != null && def.before == null) {
+                    // default is in the end
+                    newCases = loopCases(cases.slice(0), def.el.slice(0), testVar, newCases);
```

Comment:
Maybe you will replace `slice(0)` to `copy()`?

Reply:
Yeah I only adjusted the original code here - before it was `newCases = loopCases(cases.slice(0), def == null ? null : def.el.slice(0), testVar, newCases);
`
So it had before : `def == null ? null : def.el.slice(0)`

Now you'd like it to be `def.el.copy()[0]` ?

Reply:
Nope
`def.el.copy()` instead of `def.el.slice(0)`